shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Val.hs

```diff
@@ -61,11 +56,15 @@ class
   -- | Create a value containing only this amount of ADA
   inject :: Coin -> t
 
+  -- | modify the blessed Coin part of t
+  modifyCoin :: (Coin -> Coin) -> t -> t
```

Comment:
Is there particular need to have this, rather than just using `coin` and `inject`? Is this added for performance reasons?

Reply:
I want to make Value abstract. Once one has a Value there is no obvious way to modify the coin component.I suppose one could say some thing like
modify f v = f (coin v) <+> (v <-> (inject (coin v))).
But this make an O(n) instead of a constant time operation.
The wallet implementors have to construct these things, lets make it easy for them.
If we let them use the constructor of Value, they will screw up the invariant that we never store a zero quantity for sure.